deploy/examples/object-multi-instance-test.yaml

```diff
@@ -0,0 +1,107 @@
+#################################################################################################################
+# Create an object store with settings for a test environment. Only a single OSD is required in this example.
+#  kubectl create -f object-multi-instance-test.yaml
+#
+# The example below provides minimalistic test configuration for multi-instance RGW deployment.
+# It defines CephObjectRealm, CephObjectZoneGroup, and CephObjectZone CRs responsible for storage pools configuration.
+# Then, three separate CephObjectStore resources are created.
+# These stores refer to the same CephObjectZone, and therefore, to the same storage pools, representing the same ObjectStorage data.
+# The CephObjectStore resources are configured to host different RGW APIs: S3, SWIFT, and Admin-ops.
+# The configuration of the CephObjectStore can be further extended to include any number of RGW deployments backed by the same data.
+# Deployments can be customized to host different APIs, allocate different resources, use different domains,
+# and apply any other options available in the CephObjectStore CRD
+#################################################################################################################
+apiVersion: ceph.rook.io/v1
+kind: CephObjectRealm
+metadata:
+  name: my-store
+  namespace: rook-ceph # namespace:cluster
+---
+apiVersion: ceph.rook.io/v1
+kind: CephObjectZoneGroup
+metadata:
+  name: my-store
+  namespace: rook-ceph # namespace:cluster
+spec:
+  realm: my-store
+---
+apiVersion: ceph.rook.io/v1
+kind: CephObjectZone
+metadata:
+  name: my-store
+  namespace: rook-ceph # namespace:cluster
+spec:
+  zoneGroup: my-store
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 1
+      requireSafeReplicaSize: false
+  dataPool:
+    failureDomain: host
+    replicated:
+      size: 1
+      requireSafeReplicaSize: false
+    parameters:
+      compression_mode: none
+  # Alternatively, configure pool placements with pre-existing pools here.
+  # More details about Pool placements and storage classes:
+  # https://rook.io/docs/rook/latest-release/Storage-Configuration/Object-Storage-RGW/object-storage/#create-local-object-stores-with-pool-placements
+  # sharedPools:
+  #   poolPlacements:
+  #   ...
+---
+# RGW instance to host admin-ops API only
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStore
+metadata:
+  name: store-admin
```

Comment:
If my plan is right, users can consume this object store by pointing child resources to the admin. Since that is a confusing change from other object stores, I would suggest that we add a CephObjectStoreUser to this example to illustrate how to use those with this setup. That resource will also make it easy to ensure the sample is working, by checking whether the COSUser status.

IMO, it might also be reasonable to add an OBC StorageClass (not necessarily a whole OBC), also for illustration purposes, but I'll let you decide on that.

Reply:
I've added `CephObjectStoreUser` to this example and also to docs. However, when i tested it locally i noticed that generated user secret also containts `Endpoint`. And in this case it will point to admin RGW but if i got it right created user should be used for s3 api.
```shell
k get secrets rook-ceph-object-user-store-admin-multi-instance-user -o jsonpath={.data.Endpoint} | base64 -d
http://rook-ceph-rgw-store-admin.rook-ceph.svc:80
```
Is Endpoint field used?

Reply:
Sorry for the delay. Yes, the endpoint is used, or at least it's meant to be usable. Now that you've mentioned that behavior, it is obvious to me that something else is needed.

To make things integrate well, we will need to parameterize something else in the object store. My suggestion would be that we keep this PR as-is, while keeping in mind that the endpoints won't be correct for CephObStoreUser or OBCs with just this change. I would prefer to follow up with design/discussion about how to proceed for better integration. I think this change is reasonable standalone.

While I don't love the idea of having a configuration to specify the admin ops endpoint, I think that probably ends up being the simplest for users to understand, and for us to document. An alternative is to consider reusing the `advertiseEndpoint` as the inverse, but that seems a little backwards logically, and I think that would make it harder to use and explain.